### DISCUSSION TITLE
CacheConfigManager is not working

### DISCUSSION THREAD

**Discussion by RiyasIbrahim** [quarkusio/quarkus#31493] [Q&A] [answered]
I was looking into https://quarkus.io/guides/cache#quarkus-cache-config-group-cache-config-caffeine-config_configuration

I have implemeted the cache, but unable to set the expireAfterWrite at runtime. I was getting the following error
Build time property cannot be changed at runtime:
- quarkus.cache.caffeine."cache-name".expire-after-write is set to '1S' but it is build time fixed to '5M'. Did you change the property quarkus.cache.caffeine."cache-name".expire-after-write after building the application?

I think the issue is that cache is already created before the runtime values is configured, but not sure

So I was trying to implement a CacheConfigManager, but the code given in guide is wrong. What is the correct way to move forward.

Also is there a way or an option to disable cache at runtime?

[a comment by an automated account is not shown]

**Comment by gwenneg** [comment]
Hi @RiyasIbrahim. Could you please tell me the version of Quarkus you depend on?

**Reply by RiyasIbrahim** [reply]
2.15.3.Final

**Reply by gwenneg** [reply]
That explains why you are not able to change the cache configuration at runtime. That configuration could only be changed at build time until Quarkus 2.16.0.Final. If you try again with 2.16.0.Final, you should be able to change the configuration at runtime.

**Comment by gwenneg** [comment]
There is indeed in a typo in the `CacheConfigManager` code snippet. I'll fix it right away. Thanks for reporting this!

**Comment by gwenneg** [comment, marked as answer]
https://github.com/quarkusio/quarkus/pull/31633 should fix the code from the doc. Please let me know if you find any other issue.

**Reply by RiyasIbrahim** [reply]
In the version of quarkus I use, those functions are not available, is the available in latest version?

**Reply by gwenneg** [reply]
Yes, the following methods are only available starting Quarkus 2.16.0.Final:

- `CaffeineCache#setExpireAfterWrite(Duration)`
- `CaffeineCache#setExpireAfterAccess(Duration)`
- `CaffeineCache#setMaximumSize(long)`

**Reply by RiyasIbrahim** [reply]
Thanks
